common/persistence/history_manager.go

```diff
@@ -532,16 +528,23 @@ func (m *executionManagerImpl) AppendHistoryNodes(
 	request *AppendHistoryNodesRequest,
 ) (*AppendHistoryNodesResponse, error) {
 
-	req, err := m.serializeAppendHistoryNodesRequest(ctx, request)
-
+	nodeReq, err := m.serializeAppendHistoryNodesRequest(request)
 	if err != nil {
 		return nil, err
 	}
 
-	err = m.persistence.AppendHistoryNodes(ctx, req)
+	err = m.persistence.AppendHistoryNodes(ctx, nodeReq)
+	if err == nil && request.IsNewBranch {
```

Comment:
Why aren't we using the `if err != nil { return err }` pattern here?

Reply:
And below

Reply:
It did strike me as a bit odd why the `size` is returned regardless of error or not. But I believe that's why the original code also had this slight anti-pattern for error handling. I tracked this entire callstack chain and as far as I can tell, the size is expected. Hence, I'm following suit.

Reply:
I don't think we should repeat this pattern here because it sticks out like a sore thumb.